compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -230,6 +230,18 @@ impl<'a> Parser<'a> {
                 self.bump();
             }
 
+            if self.prev_token == token::BinOp(token::Plus)
+                && self.token == token::BinOp(token::Plus)
+            {
+                let op_span = self.prev_token.span.to(self.token.span);
+                // Eat the second `+`
+                self.bump();
+                // TODO: implement
+                let start_is_semi = false;
```

Comment:
I'm not sure how to figure out if the previous token is a semicolon. It seems like even at the start of this function, `prev_token` refers to the `i`.

Reply:
I can't see a way to do this in this function outside of threading some context through (which we probably don't want?) or finding somewhere else to hook this diagnostic in at.

Reply:
Any ideas what I should do?

Reply:
I think it would be best if just assume there isn't a semicolon here and downgrade the applicability of the suggestion accordingly (perhaps adjusting the message to emphasise that the user _might_ want the suggestion). Then we can land this as an improvement and revisit that case in future.

Reply:
Ok, I think I found a good workaround. Basically, emit both suggestions. However, rustfix is crashing:

```

---- [ui] ui/parser/increment-autofix.rs stdout ----
thread '[ui] ui/parser/increment-autofix.rs' panicked at 'failed to apply suggestions for "/Users/noahlev/code/rust/r1/src/test/ui/parser/increment-autofix.rs" with rustfix: Cannot replace slice of data that was already replaced', src/tools/compiletest/src/runtest.rs:3131:17
```

Any ideas?

Reply:
Can you use [`span_suggestions`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_errors/diagnostic/struct.Diagnostic.html#method.span_suggestions)?

Reply:
Ah, is that new? The only problem with it is I can't have different messages, which might be confusing.

Reply:
I think it's the only option we've got, so we'll just need to find a message that works.